atlp-rwanda/champs-ec-be
# #187300241-fix auth via google

What does this PR do?

This PR implements the necessary tasks to enable users to log in with Google OAuth2.

Description of Task to be completed:

.Implemented route for handling Google OAuth2 callback.
.Defined serialization and deserialization methods for Passport.
.Created route to initiate Google OAuth2 login.
.Returned user data and token in response after successful authentication.
.I added more testing to reach to 90%
.I refactored code to gain more readability

**Screenshots**

![Screenshot 2024-04-14 165324](https://github.com/atlp-rwanda/champs-ec-be/assets/141851169/3cc42248-ca84-4fcf-9777-6acc35a12a99)
![Screenshot 2024-04-14 165438](https://github.com/atlp-rwanda/champs-ec-be/assets/141851169/42627632-7970-434f-9d91-79e3c8f0c67c)



How should this be manually tested?

.Test the implementation by initiating the Google OAuth2 login process.
.Verify that users can successfully log in using their Google accounts.
.Ensure that user data and tokens are returned correctly upon successful authentication.
.Manually inspect the user experience to confirm the integration of Google OAuth2 login.

What are the relevant pivotal tracker/Trello stories?

Pivotal ID #187300241
